app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/AdditionalOptions.kt

```diff
@@ -0,0 +1,200 @@
+        when (additionalOptionsState) {
+            AdditionalOptionMenuState.AttachmentAndAdditionalOptionsMenu -> {
+                AttachmentAndAdditionalOptionsMenuItems(
+                    selectedOption = selectedOption,
+                    isFileSharingEnabled = isFileSharingEnabled,
+                    isEditing = isEditing,
+                    isSelfDeletingSettingEnabled = isSelfDeletingSettingEnabled,
+                    isMentionEnabled = onMentionButtonClicked != null,
+                    onMentionButtonClicked = onMentionButtonClicked ?: {},
+                    onAdditionalOptionsMenuClicked = onAdditionalOptionsMenuClicked,
+                    onGifButtonClicked = onGifOptionClicked ?: {},
+                    onSelfDeletionOptionButtonClicked = onOnSelfDeletingOptionClicked ?: {},
+                    onRichEditingButtonClicked = onRichEditingButtonClicked,
+                    onPingClicked = onPingOptionClicked
+                )
+            }
+
+            AdditionalOptionMenuState.RichTextEditing -> {
+                RichTextOptions(
+                    onRichTextHeaderButtonClicked = { onRichOptionButtonClicked(RichTextMarkdown.Header) },
+                    onRichTextBoldButtonClicked = { onRichOptionButtonClicked(RichTextMarkdown.Bold) },
+                    onRichTextItalicButtonClicked = { onRichOptionButtonClicked(RichTextMarkdown.Italic) },
+                    onCloseRichTextEditingButtonClicked = onCloseRichEditingButtonClicked
+                )
+            }
+
+            AdditionalOptionMenuState.Hidden -> {}
+        }
+    }
+}
+
+@Composable
+fun AdditionalOptionSubMenu(
+    isFileSharingEnabled: Boolean,
+    additionalOptionsState: AdditionalOptionSubMenuState,
+    tempWritableImageUri: Uri?,
+    tempWritableVideoUri: Uri?,
+    onRecordAudioMessageClicked: () -> Unit,
+    onAttachmentPicked: (UriAsset) -> Unit,
+    modifier: Modifier
+) {
+    when (additionalOptionsState) {
+        AdditionalOptionSubMenuState.AttachFile -> {
+            AttachmentOptionsComponent(
+                onAttachmentPicked = onAttachmentPicked,
+                tempWritableImageUri = tempWritableImageUri,
+                tempWritableVideoUri = tempWritableVideoUri,
+                isFileSharingEnabled = isFileSharingEnabled,
+                onRecordAudioMessageClicked = onRecordAudioMessageClicked,
+                modifier = modifier
+            )
+        }
+
+        AdditionalOptionSubMenuState.RecordAudio -> {
+            RecordAudioComponent(
+                onAudioRecorded = {},
+                modifier = modifier
+            )
+        }
+        // non functional for now
+        AdditionalOptionSubMenuState.AttachImage -> {}
+        AdditionalOptionSubMenuState.Emoji -> {}
+        AdditionalOptionSubMenuState.Gif -> {}
+        AdditionalOptionSubMenuState.Hidden -> {}
+    }
+}
+
+@Composable
+fun RecordAudioComponent(
+    onAudioRecorded: () -> Unit,
+    modifier: Modifier
```

Comment:
Actually this component was created as a base for my work on sending audio messages, which is being worked on right now branched out of this PR so I suppose at least this, it can be ignored